File: order.service/src/main/java/br/com/edu/order/infra/rest/v1/response/Page.java
package br.com.edu.order.infra.rest.v1.response;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Data;

@Data
@AllArgsConstructor
public class Page {

    @JsonProperty("page")
    private Integer page;

    @JsonProperty("page_size")
    private Integer pageSize;

    @JsonProperty("total_pages")
    private Integer totalPages;

    @JsonProperty("total_elements")
    private Long totalElements;

}

File: order.service/order-service-app/src/main/kotlin/order/infra/rest/v1/response/Page.kt
package order.infra.rest.v1.response

import com.fasterxml.jackson.annotation.JsonInclude
import com.fasterxml.jackson.annotation.JsonProperty

@JsonInclude(JsonInclude.Include.NON_NULL)
data class Page(

    @JsonProperty("page")
    val page: Int = 0,

    @JsonProperty("page_size")
    val pageSize: Int = 0,

    @JsonProperty("total_pages")
    val totalPages: Int = 0,

    @JsonProperty("total_elements")
    val totalElements: Long = 0,
)